src/utils.ts

```diff
@@ -0,0 +1,138 @@
+import { SchemaProperty } from './components/form-element'
+
+export interface Conditional {
+    compiled: string
+    then?: SchemaProperty
+    else?: SchemaProperty
+}
+
+//checks if item is object
+export const isObject = (item: any) => {
+    return item && typeof item === 'object' && !Array.isArray(item)
+}
+
+//merges the current object at all nesting levels with a new one
+//the method does not return a new object; the current one is being altered
+export const addProperties = (currentObject: any, newProperties: any): any => {
+    if (isObject(currentObject) && isObject(newProperties)) {
+        for (const key in newProperties) {
+            if (isObject(newProperties[key])) {
+                if (!currentObject[key]) {
+                    Object.assign(currentObject, { [key]: {} })
+                }
+                addProperties(currentObject[key], newProperties[key])
+            } else {
+                if(key === 'required' && Array.isArray(newProperties[key]) && Array.isArray(currentObject[key])){
+                    //@ts-ignore
+                    currentObject[key] = [... new Set(currentObject[key].concat(...newProperties[key]))]
+                } else {
+                    Object.assign(currentObject, { [key]: newProperties[key] })
+                }
+            }
+        }
+    }
+}
+
+//gets all conditionals specified at root level of a schema;
+//conditionals that are both specified with simple ifs, or as a list with allOf, anyOf, oneOf
+export const getConditionals = (schema: SchemaProperty): Conditional[] => {
+    let ifEntries = []
+    const simpleConditional = getSimpleConditional(schema)
+    if (simpleConditional) {
+        ifEntries.push(simpleConditional)
+    }
+    ifEntries = ifEntries.concat(getMultipleConditionals(schema))
+    return ifEntries.map((ifEntry) => getCompiledConditional(ifEntry))
+}
+
+//given a schema, the function will return a list with all the paths of the schema and the value of the leaf node
+//keywords such as properties and const are not included
+//usage: extract all the conditions specified inside an if statement
+const getPropertyPaths = (schema: any, path = ''): { '0': string; '1': any }[] => {
+    if (isObject(schema)) {
+        return Array.prototype.concat.apply(
```

Comment:
`return [...Object.keys(schema).map((key) =>
                getPropertyPaths(schema[key], key === 'properties' || key === 'const' ? path : path + '.' + key)
            )]`
?

Reply:
I want to get the data paths of all properties in an if statement; the function has to ignore the properties layer of the schema (it does not exist on the data object), as well as the const keyword; they will not be appendend to the path;
e.g. the following schema given as parameter 

{
          "properties": {
                "second": {
                    "const": true
                 }
           }
 }

will return

[{'second': true}]